### PR TITLE
ItemCollection_Refactor

### LINKED ISSUE #424: General refactoring
### Current Behaviour:
A lot of functions have many if-else branches and the procedures implemented by these branches have too much code in them. This makes readability and testability hard.

For example the branch in this [chunk of code](https://github.com/HTTP-APIs/hydrus/blob/42205ab4d4984f591310b19f3b5d6c1610e95e4d/hydrus/resources.py#L108) should be ported into a function like:
```
    if checkClassOp(class_path, "GET"):
        items_get_check_support() 
    abort(405)
```
Into a dedicated module called `itemshelpers.py` the above functions should be defined by incorporating the current code:
```
def items_get_check_support(...):
    """ Check if class_type supports GET operation
            try:
                # Try getting the Item based on ID and Class type
                response = crud.get(
                    id_,
                    class_type,
                    api_name=get_api_name(),
                    session=get_session())

                response = finalize_response(class_path, response)
                return set_response_headers(
                    jsonify(hydrafy(response, path=path)))

            except (ClassNotFound, InstanceNotFound) as e:
                error = e.get_HTTP()
                return set_response_headers(jsonify(error.generate()), status_code=error.code)
```
Each of these helpers functions should be tested separately.

### Expected Behaviour:
All functions should perform well-scoped units of work and have their own tests.
Handlers in endpoints classes should be much more concise, all the units of work for each branch should be moved into helpers functions. 

Main functions should look something like this:
```
def post(...):
    """ Docstring """
    ... some code ...
    if condition_1:
        run_helper_1(...)
    elif condition_2:
        run_helper_2(...)
    else:
        run_helper_3(...)
    ... rest of the code ...
```

### PR DESCRIPTION
**Pull request by thrishik7**
Fixes #424 

### Checklist
- [x] My branch is up-to-date with upstream/develop branch.
- [x] Everything works and tested for Python 3.5.2 and above.

### Description
formatted docstring

### CODE REVIEW

**Comment by thrishik7** [comment]
@xadahiya I have formatted the docstrings to `itemcollection_get_support` format. If you find anything to change pls inform

File: hydrus/itemhelpers.py [hunk 1]
```diff
@@ -155,3 +164,160 @@ def items_delete_check_support(id_, class_type, path):
         error = e.get_HTTP()
         return set_response_headers(jsonify(error.generate()),
                                     status_code=error.code)
+
+
+def itemsCollection_get_support(collection, class_path, path, search_params):
```
Comment (chrizandr) [inline hydrus/itemhelpers.py]: These functions are not really checking. They perform CRUD operations. The docstrings need to be updated.

**Comment by thrishik7** [comment]
@chrizandr  updated docstrings

**Comment by xadahiya** [comment]
Please resolve conflicts.

**Comment by sameshl** [comment]
I think this PR can be closed as the refactoring here is already merged in this PR https://github.com/HTTP-APIs/hydrus/pull/459.

**Pull request closed without merging**